src/python_testing/TC_OCC_3_1.py

```diff
@@ -0,0 +1,127 @@
+            # write 10 as a HoldTime attibute
+            write_res = await ChipDeviceCtrl.WriteAttribute(node_id, [(endpoint, attributes.HoldTime(hold_time))])
+            asserts.assert_equal(write_res[0].status, Status.Success, "Write HoldTime failed")
+
+        else:
+            logging.info("No HoldTime attribute supports. Will test only occupancy attribute triggering functionality")
+            
+        self.step(3)
+        # check if Occupancy attribute is 0
+        occupancy_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.Occupancy)
+
+        # if occupancy is on, then wait until the sensor occupancy state is 0.
+        if occupancy_dut == 1:
+            # Don't trigger occupancy sensor to render occupancy attribute to 0
+            if attributes.HoldTime.attribute_id in attribute_list:
+                time.sleep(hold_time + 2) # add some extra 2 seconds to ensure hold time has passed.
+            else: # a user wait until a sensor specific time to change occupancy attribute to 0.  This is the case where the sensor doesn't support HoldTime.
+                self.wait_for_user_input(
+                    prompt_msg="Type any letter and press ENTER after the sensor occupancy is detection ready state (occupancy attribute = 0)")
+
+        # check sensor occupancy state is 0 for the next test step
+        occupancy_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.Occupancy)
+        asserts.assert_equal(occupancy_dut, 0, "Occupancy attribute is still 1.")
+
+        self.step(4)
+        # Trigger occupancy sensor to change Occupancy attribute value to 1 => TEST ACTION on DUT
+        self.wait_for_user_input(prompt_msg="Type any letter and press ENTER after a sensor occupancy is triggered.")
+
+        # And then check if Occupancy attribute has changed.
+        occupancy_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.Occupancy)
+        asserts.assert_equal(occupancy_dut, 1, "Occupancy state is not changed to 1")
+        
+        self.step(5)
+        # check if Occupancy attribute is back to 0 after HoldTime attribute period
+        if attributes.HoldTime.attribute_id in attribute_list:
+            # message to the tester
+            self.wait_for_user_input(prompt_msg="Do not trigger sensor during HoldTime.")
+            
+            # Start a timer based on HoldTime
+            time.sleep(hold_time+2) # add some extra 2 seconds to ensure hold time has passed.
+        
+            occupancy_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.Occupancy)
+            asserts.assert_equal(occupancy_dut, 0, "Occupancy state is not 0 after HoldTime period")
+            
+        else:
+            logging.info("HoldTime attribute not supported. Skip this test procedure.")
+            self.skip_step(5)

```

Comment:
in other places you have `mark_current_step_skipped` ... what is the difference between using this vs the other?

Reply:
skip_step needs to provide test step number.  But for mark_current_step_skipped, you don't provide the number.  It just skip the current test step you are in.  You can find it detail from matter_testing_support.py